docs/index.md

```diff
@@ -6,69 +6,87 @@ This page is the top level entry point into the Substrate documentation.
 
 > **Some Expertise Needed**
 >
->In order to get the most out of Substrate, you should have a good knowledge of blockchain concepts
->and basic cryptography. Terminology like header, block, client, hash, transaction and signature
->should be familiar. At present you will need a working knowledge of Rust to be able to do any
->significant customization/adaption of Substrate (though eventually, we aim for this not to be the
->case).
+> In order to get the most out of Substrate, you should have a good knowledge of blockchain concepts and basic
+> cryptography. Terminology like header, block, client, hash, transaction and signature should be familiar. At present
+> you will need a working knowledge of Rust to be able to do any significant customization/adaption of Substrate (though
+> eventually, we aim for this not to be the case).
 
 Substrate is a blockchain development framework with a completely generic State Transition Function
-([STF](knowledgebase/getting-started/glossary#stf-state-transition-function)) and modular components for consensus, networking,
-and configuration.
-
-Despite being "completely generic", it comes with both standards and conventions - particularly with
-the Substrate runtime module library (a.k.a [FRAME](knowledgebase/runtime/frame.md)) - regarding the underlying
-data-structures that power the STF, thereby making rapid blockchain development a reality.
+([STF](knowledgebase/getting-started/glossary#stf-state-transition-function)) and modular components for consensus,
+networking, and configuration.
+
+Despite being "completely generic", it comes with both standards and conventions - particularly with the Substrate
+runtime module library (a.k.a [FRAME](knowledgebase/runtime/frame.md)) - regarding the underlying data-structures that
+power the STF, thereby making rapid blockchain development a reality.
+
+## Architecture
+
+![Substrate Client Architecture](assets/substrate-arch.png)
+
+The Substrate client is an application that runs a Substrate-based blockchain node - it consists of several components
+that include, but are not limited to, the following:
+
+- **Storage** is used to persist the evolving state of the decentralized system represented by a blockchain. The
+  blockchain network allows participants to reach trustless [consensus](knowledgebase/advanced/consensus) about the
+  state of storage. Substrate ships with a simple and highly efficient
+  [key-value storage mechanism](knowledgebase/advanced/storage).
+- **Runtime** logic defines how blocks are processed, including state transition logic. In Substrate, runtime code is
```

Comment:
The emphasis I would put on the substrate client is that it contains a **Wasm interpreter** which is used to execute the runtime. Either an additional bullet point above this about the wasm interpreter or in replacement of this line makes sense to me.

Reply:
This is one of the sections I struggled with the most. You are undoubtedly technically correct, but I'm not sure if what you're describing is the most constructive abstraction to create for our users. For one thing, the interpreter is not included in the diagram and I'm trying to keep this list of components restricted to what appears in the diagram. I also think defining the concept of a runtime and describing that, in Substrate, it is compiled to a Wasm blob and included as part of the chain state, is more important than the executor/interpreter, which I see as more of an implementation/optimization detail.

Reply:
The "substrate client" is the executable that runs a node.

The wasm interpreter included with it is certainly more important than the included runtime. You can pass to your substrate client a chain specification with a runtime that is different than the one included in the client, and it will execute that wasm runtime in the chainspec instead of the one compiled into the client.

For anyone to write a Substrate/Polkadot client in another language would not need to include the runtime information into the client, just provide a Wasm executor.

The runtime being included into the client is the optimization, and optional piece.

Reply:
> The wasm interpreter included with it is certainly more important than the included runtime.

I think this is where we disagree.

Reply:
I am not sure where to go from here. You should ask for feedback in the public channels and get other people's opinion then.

Take a look at the `client` folder in Substrate: https://github.com/paritytech/substrate/tree/master/client

Reply:
Let's see what the other members of the Developer Hub team think

Reply:
```suggestion
- **Runtime Executor**  executed the runtime logic which defines how blocks are processed, including state transition logic. In Substrate, runtime code is
```

Here's an attempt at a compromise. I think Shawn's point is valid ad the executor is worth mentioning. I also think Dan's point about it being too fine of a detail to fit what he's explaining here is valid. Ultimately I think this PR is a big improvement, and I'm fine with it going in as is, with my suggestion, or with some other rewording.

Reply:
I added two sentences to the end of this bullet point that I hope covers all our bases in a way that is complete _and_ easy to understand https://github.com/substrate-developer-hub/substrate-developer-hub.github.io/pull/619/commits/9ad51ffe215f4fd8c03c77b6c0e371f6b548f914 It's a little on the verbose side but I think it has the nice "side effect" of adding color to the forkless runtime comment.